File: algorithms/string reversal/reverseString.js
function iterativeReverseString(str) {
  let reversed = "";
  for (let i = 0; i < str.length; i++) {
    reversed += str[str.length - i - 1];
  }
  return reversed;
}

function recursiveReverseString(str) {
  if (str.length === 0) {
    return str;
  } else {
    return recursiveReverseString(str.substr(1)) + str[0];
  }
}

const reversed = iterativeReverseString("reverse");
const palindrome = iterativeReverseString("racecar");
console.log(reversed, palindrome);

const reversed2 = recursiveReverseString("reverse");
const palindrome2 = recursiveReverseString("racecar");
console.log(reversed2, palindrome2);

File: algorithms/string reversal/reverse_string.py
def iterative_reverse_string(str):
    result = ""
    for i in range(len(str)-1, -1, -1):
        result += str[i]
    return result


def recursive_reverse_string(str):
    if len(str) == 0:
        return str
    else:
        return recursive_reverse_string(str[1:]) + str[0]



reverse_string1 = iterative_reverse_string("reverse")
palindrome1 = iterative_reverse_string("racecar")
print(reverse_string1, palindrome1)

reverse_string2 = recursive_reverse_string("reverse")
palindrome2 = recursive_reverse_string("racecar")
print(reverse_string2, palindrome2)